### PR TITLE
Fixed not being able to detect content emptyness

### PR DESCRIPTION
**Pull request by Devorein**
We couldn't detect page content emptiness if there was a non-content node. See below (a page node). I encountered this while testing out the notion import
image
1. Correctly detect content emptiness for non-content nodes (mention, pages, cryptoPrice and so on)
2. Show the correct icon in the sidebar if the page is empty (couldn't detect correctly earlier)

### CODE REVIEW

File: components/common/PageLayout/components/PageNavigation.tsx [hunk 1]
```diff
@@ -490,9 +491,7 @@ function RenderDraggableNode ({ item, onDropAdjacent, onDropChild, pathPrefix, a
 
   const { focalboardViewsRecord } = useFocalboardViews();
 
-  const docContent = (item.content as PageContent)?.content;
-  const isEmptyContent = docContent && (docContent.length <= 1
-    && (!docContent[0] || (docContent[0] as PageContent)?.content?.length === 0));
+  const isEmptyContent = checkForEmpty(item.content as PageContent);
```
Comment (mattcasey) [inline components/common/PageLayout/components/PageNavigation.tsx:494]: remember item.content can be null too, not sure you need to cast it here

File: components/common/CharmEditor/CharmEditor.tsx [hunk 2]
```diff
@@ -278,10 +284,7 @@ function CharmEditor (
 
   function _onContentChange (view: EditorView) {
     // @ts-ignore missing types from the @bangle.dev/react package
-    const docContent: { content: { size: number } }[] = view.state.doc.content.content;
```
Comment (mattcasey) [inline components/common/CharmEditor/CharmEditor.tsx:281]: i am pretty sure thtis is is a different structure than what we write to our page

**Review by mattcasey — APPROVED** [review]

File: components/common/PageLayout/components/PageNavigation.tsx (hunk 1 above)
Comment (Devorein) [reply components/common/PageLayout/components/PageNavigation.tsx:494]: Yeah you are right. Thanks for the edit

File: components/common/CharmEditor/CharmEditor.tsx (hunk 2 above)
Comment (Devorein) [reply components/common/CharmEditor/CharmEditor.tsx:281]: This is the structure before converting to json. We are converting it to json before passing to the emptiness detection function

**Pull request merged**